static/js/components/widgets/PeopleSelector.js

```diff
@@ -0,0 +1,186 @@
+// @flow
+import React from "react"
+import R from "ramda"
+import Autosuggest from "react-autosuggest"
+import { arrayMove } from "react-sortable-hoc"
+import { connect } from "react-redux"
+
+import PeopleList from "./PeopleList"
+import { PeopleItem } from "./PeopleItem"
+import SearchTextbox from "../SearchTextbox"
+
+import { actions } from "../../actions"
+import { searchResultToProfile } from "../../lib/search"
+import { SEARCH_FILTER_PROFILE } from "../../lib/picker"
+
+import type { Dispatch } from "redux"
+import type { Profile } from "../../flow/discussionTypes"
+
+type Props = {
+  clearSearch: () => void,
+  profiles: Array<Profile>,
+  updateProfiles: (newJson: Array<Profile>) => void,
+  suggestions: Array<Profile>,
+  fetchSuggestions: (text: string) => Promise<Array<Profile>>
+}
+type State = {
+  text: string
+}
+export class PeopleSelector extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      text: ""
+    }
+  }
+
+  componentDidMount() {
+    const { clearSearch } = this.props
+    clearSearch()
+  }
+
+  onSuggestionsFetchRequested = ({
+    value,
+    reason
+  }: {
+    value: string,
+    reason: string
+  }) => {
+    const { clearSearch, fetchSuggestions } = this.props
+    if (reason !== "input-changed") {
+      return
+    }
+
+    if (value.trim().length === 0) {
+      clearSearch()
+      return
+    }
+
+    // NOTE: this may cause a race condition if a user types quickly and some requests resolve out of order
+    fetchSuggestions(value)
+  }
+
+  renderInputComponent = (inputProps: Object) => (
+    <SearchTextbox {...inputProps} />
+  )
+
+  onChangeText = (
+    event: Event,
+    { newValue, method }: { newValue: string, method: string }
```

Comment:
Same feedback here as the one above, might be worth throwing them in their own file under `js/flow`. Looks like there's a few other cases of this but I won't comment on every one.

Reply:
What do you mean by other cases?